source/adapters/level_zero/event.cpp

```diff
@@ -145,6 +145,10 @@ ur_result_t urEnqueueEventsWait(
     std::unique_lock<ur_shared_mutex> Lock(Queue->Mutex);
     resetCommandLists(Queue);
   }
+  if (OutEvent && (*OutEvent)->Completed) {
```

Comment:
Can you add a comment about why this is needed and why this does not get cleaned up later?